internal/engine/filter/filter.go

```diff
@@ -214,6 +230,14 @@ func (c Criteria) ToGmailSearch() string {
 	return strings.Join(res, " ")
 }
 
+// ToGmailSearchURL returns the equivalent query in an URL to Gmail search.
+func (c Criteria) ToGmailSearchURL() string {
```

Comment:
As we're at it, could you add just a couple of tests checking the output format?